clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp

```diff
@@ -222,60 +222,145 @@ class matmul_desc_t {
 
 namespace detail {
 /// Sacling each row of matrix D with the corresponding element of vector alpha.
-template <class T, class Talpha>
-sycl::event scale_d_with_vector_alpha_impl(::dpct::cs::queue_ptr q_ptr,
-                                           int rows, int cols, T *d,
-                                           const Talpha *alpha,
-                                           std::vector<sycl::event> deps) {
+template <class T, class Tscale>
+sycl::event scale_new_a_impl(::dpct::cs::queue_ptr q_ptr, int rows, int cols,
+                             T *a, const Tscale *alpha, bool vector_alpha,
+                             bool device_alpha, const Tscale *a_scale,
+                             const Tscale *b_scale,
+                             std::vector<sycl::event> deps) {
+  if (device_alpha) {
+    return q_ptr->submit([&](sycl::handler &cgh) {
+      cgh.depends_on(deps);
+#ifdef DPCT_USM_LEVEL_NONE
+      access_wrapper<T *> a_acc(a, cgh);
+      const Tscale *alpha_ptr = nullptr;
+      if (device_alpha)
+        alpha_ptr = alpha;
+      access_wrapper<const Tscale *> alpha_acc(alpha_ptr, cgh);
+      auto a_scale_buf = a_scale ? dpct::get_buffer(a_scale) : dpct::buffer_t(sycl::range<1>(0));
```

Comment:
use this way to pass the alpha to device side. Then we can merge the 2 submit calls.